mailbox/store/src/main/java/org/apache/james/mailbox/store/search/comparator/SentDateComparator.java

```diff
@@ -18,44 +18,69 @@
  ****************************************************************/
 package org.apache.james.mailbox.store.search.comparator;
 
-import java.io.StringReader;
+import java.time.Instant;
+import java.time.ZonedDateTime;
 import java.util.Comparator;
-import java.util.Date;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 import org.apache.james.mailbox.store.mail.model.MailboxMessage;
-import org.apache.james.mime4j.dom.datetime.DateTime;
-import org.apache.james.mime4j.field.datetime.parser.DateTimeParser;
-import org.apache.james.mime4j.field.datetime.parser.ParseException;
+import org.apache.james.util.date.ImapDateTimeFormatter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.annotations.VisibleForTesting;
 
 /**
  * {@link Comparator} which works like stated in RFC5256 2.2 Sent Date
- *
  */
 public class SentDateComparator extends AbstractHeaderComparator {
+
     public final static Comparator<MailboxMessage> SENTDATE = new SentDateComparator();
+    private static final Logger LOGGER = LoggerFactory.getLogger(SentDateComparator.class);
+    // Some sent e-mail have this form : Wed,  3 Jun 2015 09:05:46 +0000 (UTC)
+    // Java 8 Time library RFC_1123_DATE_TIME corresponds to Wed,  3 Jun 2015 09:05:46 +0000 only
+    // This REGEXP is here to match ( in order to remove ) the possible invalid end of a header date
+    // Example of matching patterns :
+    //  (UTC)
+    //  (CEST)
+    private static final Pattern DATE_SANITIZING_PATTERN = Pattern.compile(" *\\(.*\\) *");
+
+    public static Optional<ZonedDateTime> toISODate(String value) {
```

Comment:
That code is moved from ES to avoid circular dependency and share same code path.

Maybe this code is orthogonal with the issue.